torch/distributed/rpc/api.py

```diff
@@ -163,8 +164,8 @@ def _all_gather(obj):
         _all_gather_sequence_id += 1
 
     is_leader = leader_name == self_name
-    # Set a long enough timeout for all shutdown messages to be processed.
-    timeout = 5  # second
+    if timeout == UNSET_RPC_TIMEOUT:
+        timeout = _get_current_rpc_agent()._get_timeout()
```

Comment:
If we can use it, `set_timeout = rpc.get_rpc_timeout()` might read simpler here.